test/transactions/zap-repay.ts

```diff
@@ -0,0 +1,132 @@
+  });
+
+  snapshotAndRevertEach();
+
+  it('user zap repay USDC in USDC market', async function () {
+    await claimToken(chainId, user.address, baseToken, '200');
+
+    // 1. user obtains a quotation for zap repay 100 USDC through the zap repay API
+    const srcToken = polygonTokens.USDC;
+    const srcAmount = '100';
+    const slippage = 100;
+    const permit2Type = 'approve';
+    const quotation = await api.quote(
+      chainId,
+      marketId,
+      'zap-repay',
+      {
+        account: user.address,
+        srcToken,
+        srcAmount,
+        slippage,
+      },
+      permit2Type
+    );
+
+    // 2. user needs to allow the Protocolink user agent to repay on behalf of the user
+    expect(quotation.approvals.length).to.eq(2);
+    for (const approval of quotation.approvals) {
+      await expect(user.sendTransaction(approval)).to.not.be.reverted;
+    }
+
+    // 3. user obtains a zap repay transaction request through the build transaction API.
+    expect(quotation.logics.length).to.eq(1);
+    const transactionRequest = await api.buildRouterTransactionRequest({
+      chainId,
+      account: user.address,
+      logics: quotation.logics,
+    });
+
+    await expect(user.sendTransaction(transactionRequest)).to.not.be.reverted;
+
+    // 4. user's USDC borrow balance should decrease.
+    // 4-1. supply grows when the block of getting api data is different from the block of executing tx
+    const quoteDestAmount = new common.TokenAmount(baseToken, quotation.quotation.destAmount);
+    const [min] = utils.bpsBound(quoteDestAmount.amount);
+    const minDestAmount = quoteDestAmount.clone().set(min);
+
+    const borrowBalance = await service.getBorrowBalance(marketId, user.address, baseToken);
+    expect(initBorrowBalance.clone().sub(borrowBalance).lte(quoteDestAmount)).to.be.true;
+    expect(initBorrowBalance.clone().sub(borrowBalance).gte(minDestAmount)).to.be.true;
+
+    // 5. user's USDC balance should decrease
+    await expect(user.address).to.changeBalance(baseToken, -srcAmount);
+  });
+
+  it('user zap repay extra USDT in USDC market', async function () {
+    await claimToken(chainId, user.address, polygonTokens.USDT, '80000');
+    await claimToken(chainId, user.address, polygonTokens.MATIC, '500');
+
+    // 1. user obtains a quotation for zap withdraw 0.1 WETH to USDT through the zap withdraw API
+    const srcToken = polygonTokens.USDT;
+    const srcAmount = '80000';
+    const slippage = 100;
+    const permit2Type = 'approve';
+    const quotation = await api.quote(
+      chainId,
+      marketId,
+      'zap-repay',
+      {
+        account: user.address,
+        srcToken,
+        srcAmount,
+        slippage,
+      },
+      permit2Type
+    );
+
+    const quoteDestAmount = new common.TokenAmount(baseToken, quotation.quotation.destAmount);
+    const srcTokenAmount = new common.TokenAmount(srcToken, srcAmount);
+    expect(quoteDestAmount.lt(srcTokenAmount)).to.be.true;
```

Comment:
What is the purpose of comparing the amount of these two different tokens?

Reply:
This ensures that the repaid value is smaller than the source value which is the case we want to test. (we don't include the token price because `srcToken` is a stablecoin)

Reply:
will get user's debt position and use it as a base to calculate the `srcTokenAmount`